README.md

```diff
@@ -1,17 +1,119 @@
-# Griptape Tool Template
+# Griptape Extension Template
 
-A Github template repository for creating Griptape tools.
+A Github template repository for creating Griptape extensions.
 
-## Usage
+## Getting Started
 
-This template repository defines the typical structure for a Griptape tool and is meant to be used as a starting point when developing new tools. To create a tool using this template, click the **"Use this template"** button at the top of the repository.
+Via github web page:
 
-### Steps for creating a new Griptape tool:
+Click on `Use this template`
+
+![](https://docs.github.com/assets/cb-36544/images/help/repository/use-this-template-button.png)
+
+
+Via `gh`:
+
+```
+$ gh repo create griptape-extension-name -p griptape/griptape-extension-template
+```
+
+## What is a Griptape Extension?
+
+Griptape Extensions can add new functionality to the [griptape framework](https://github.com/griptape-ai/griptape), such as new Tools, Drivers, Tasks, or Structures.
+
+This repository provides a recommended structure for organizing your extension code, as well as helpful tools for testing and development.
+
+## Extension Structure
+
+The template repository is structured as follows:
+
+```bash
+tree -I __init__.py -I __pycache__
+
+├── griptape
+│   └── extension_name # Name whatever you want
+│       └── tools
+│           └── reverse_string
+│               └── tool.py
+            ...more directories for other interfaces (drivers, tasks, structures, etc)...
+└── tests
+    └── unit
+        └── tools
+            └── test_reverse_string_tool.py
+├── examples
+│   └── example.py # Example usage of the extension 
+├── LICENSE # Choose the appropriate license
+├── Makefile # Contains useful commands for development
+├── pyproject.toml # Contains the project's metadata
+├── README.md #  Describes the extension and how to use it
+```
+
+## Development
+
+### Poetry
+
+This project uses [Poetry](https://python-poetry.org/) for dependency management.
+It is recommended to configure Poetry to use [in-project](https://python-poetry.org/docs/configuration/#virtualenvsin-project) virtual environments:
+
+```bash
+poetry config virtualenvs.in-project true
+```
+
+This will create a `.venv` directory in the project root, where the virtual environment will be stored.
+This ensures that the virtual environment is always in the same location, regardless of where the project is cloned.
+
+### Useful Commands
+
+#### Installing Dependencies
+
+```bash
+make install
+```
+
+#### Running Tests
+
+```bash
+make test
+```
+
+#### Running Checks (linting, formatting, etc)
+
+```bash
+make check
+```
+
+#### Running Formatter
+
+```bash
+make format
+```
+
+#### Running Example
+
```

Comment:
1. We gave you an example you can run
2. Here's what it does in plain English
3. Here's what role the plugin/extension plays in that experience
4. Here's where to look for when you want to learn how the plugin/extension was built and hooked up